lib/widgets/profile.dart

```diff
@@ -0,0 +1,261 @@
+        // That's readable as is, but:
+        // TODO format this date using user's locale.
+        return _buildTextWidget(value);
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+        return _buildTextWidget(value);
+      case CustomProfileFieldType.unknown:
+        return null;
+    }
+  }
+
+  Widget? _buildChoiceField(BuildContext context, String value, CustomProfileField realmField) {
+    // TODO(server): This isn't really documented.  But see chat thread:
+    //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1383005
+    final CustomProfileFieldChoices choiceFieldData;
+    try {
+      final realmFieldData = jsonDecode(realmField.fieldData);
+      choiceFieldData = CustomProfileFieldChoices.fromJson(realmFieldData);
+    } on FormatException {
+      return null;
+    } on TypeError {
+      return null;
+    }
+    final choiceItem = choiceFieldData.choices[value];
+    if (choiceItem == null) return null;
+    return _buildTextWidget(choiceItem.text);
+  }
+
+  Widget? _buildExternalAccountField(BuildContext context, String value, CustomProfileField realmField) {
+    // TODO(server): This is undocumented.  See chat thread:
+    //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/external.20account.20custom.20profile.20fields/near/1387213
+    final store = PerAccountStoreWidget.of(context);
+    final CustomProfileFieldExternalAccount externalAccountFieldData;
+    try {
+      final realmFieldData = jsonDecode(realmField.fieldData);
+      externalAccountFieldData = CustomProfileFieldExternalAccount.fromJson(realmFieldData);
+    } on FormatException {
+      return null;
+    } on TypeError {
+      return null;
+    }
+    final urlPattern = externalAccountFieldData.urlPattern ??
+      store.realmDefaultExternalAccounts[externalAccountFieldData.subtype]?.urlPattern;
+    if (urlPattern == null) return null;
+    final url = urlPattern.replaceFirst("%(username)s", value);
+    return _buildLinkWidget(context, url, value);
+  }
+
+  Widget? _buildUserField(BuildContext context, String value, CustomProfileField realmField) {
+    // TODO(server): This is completely undocumented.  The key to
+    //   reverse-engineering it was:
+    //   https://github.com/zulip/zulip/blob/18230fcd9/static/js/settings_account.js#L247
+    final List<int> userIds;
+    try {
+      userIds = (jsonDecode(value) as List<dynamic>).map((e) => e as int).toList();
+    } on FormatException {
+      return null;
+    } on TypeError {
+      return null;
+    }
+    if (userIds.isEmpty) {
+      return null;
+    }
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: userIds.map((userId) => _buildUserWidget(context, userId)).toList());
+  }
+
+  Widget _buildTextWidget(String text) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 4),
```

Comment:
This feels vertically pretty cramped to me:
![image](https://github.com/zulip/zulip-flutter/assets/28173/0579c3e2-5929-455f-9549-f3d7be29bbb2)

Compare zulip-mobile:
![image](https://github.com/zulip/zulip-flutter/assets/28173/20580cc7-e330-494e-9817-7fd3e6dc5027)

(The difference in URL text is due to the bug I mentioned in a comment above where the arguments to `_buildLinkWidget` are swapped.)

I see that the source code in both implementations appears to call for the same 8px of spacing in between, though. Hmm — I suspect this `TextStyle.height` setting is the issue:
```dart
      style: const TextStyle(fontSize: 15, height: 1),
```
Aha, yep, if I remove both that and the `height: 1.5` elsewhere, it gets more spacious. I think a typical `TextStyle.height` is more like 1.2, not 1.0.

The best solution here may be to just make less use of `DefaultTextStyle` — or at least to not use it for `TextStyle.height`. In particular the upper `DefaultTextStyle` is only applying to three Text widgets that I see, plus to `_ProfileDataTable` which overrides most of it; and of those three, one is a button label, which seems like it shouldn't necessarily get the same styles as the information text anyway.

Reply:
I'll remove it from the profile data table, in the refactor away from the `Table` widget I'll have the `Row` builders apply whatever style to the labels. I think it's still useful to have it on the upper part, although it only has three `Text` widgets currently, there will be more fields to be added later when the features backing them are introduced.

Reply:
Cool, that seems like a reasonable strategy. Though it looks like in the revision, you ended up doing away with the `DefaultTextStyle` entirely, in favor of having style constants (on a container class) and referring to those directly at `Text` widgets, and that approach looks good to me.